Source/control.cpp

```diff
@@ -112,14 +112,14 @@ char SpellITbl[MAX_SPELLS] = {
 	39, 42, 41, 40, 10, 36, 30
 };
 int PanBtnPos[8][5] = {
-	{ 9, 361, 71, 19, 1 },
-	{ 9, 387, 71, 19, 0 },
-	{ 9, 427, 71, 19, 1 },
-	{ 9, 453, 71, 19, 0 },
-	{ 560, 361, 71, 19, 1 },
-	{ 560, 387, 71, 19, 0 },
-	{ 87, 443, 33, 32, 1 },
-	{ 527, 443, 33, 32, 1 }
+	{ 9 + WIDTH_DIFF_2, 361 + HEIGHT_DIFF, 71, 19, 1 },
+	{ 9 + WIDTH_DIFF_2, 387 + HEIGHT_DIFF, 71, 19, 0 },
+	{ 9 + WIDTH_DIFF_2, 427 + HEIGHT_DIFF, 71, 19, 1 },
+	{ 9 + WIDTH_DIFF_2, 453 + HEIGHT_DIFF, 71, 19, 0 },
+	{ 560 + WIDTH_DIFF_2, 361 + HEIGHT_DIFF, 71, 19, 1 },
+	{ 560 + WIDTH_DIFF_2, 387 + HEIGHT_DIFF, 71, 19, 0 },
+	{ 87 + WIDTH_DIFF_2, 443 + HEIGHT_DIFF, 33, 32, 1 },
+	{ 527 + WIDTH_DIFF_2, 443 + HEIGHT_DIFF, 33, 32, 1 }

```

Comment:
This causes the game to crash for me:
```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==17337==ERROR: AddressSanitizer: SEGV on unknown address 0x6320004e77c9 (pc 0x55be305151e2 bp 0x7fffcaaeb110 sp 0x7fffcaaeb0c0 T0)
==17337==The signal is caused by a READ memory access.
    #0 0x55be305151e1 in dvl::DrawPanelBox(int, int, int, int, int, int) /home/ajenbo/code/diablo/devilutionX/Source/control.cpp:498
    #1 0x55be305192ad in dvl::DrawCtrlPan() /home/ajenbo/code/diablo/devilutionX/Source/control.cpp:757
    #2 0x55be30475f22 in dvl::DrawAndBlit() /home/ajenbo/code/diablo/devilutionX/Source/scrollrt.cpp:1111
    #3 0x55be2fed5bc9 in dvl::run_game_loop(unsigned int) /home/ajenbo/code/diablo/devilutionX/Source/diablo.cpp:138
    #4 0x55be2fed5a5d in dvl::StartGame(int, int) /home/ajenbo/code/diablo/devilutionX/Source/diablo.cpp:113
    #5 0x55be30101c6d in dvl::mainmenu_init_menu(int) /home/ajenbo/code/diablo/devilutionX/Source/mainmenu.cpp:145
    #6 0x55be30101c31 in dvl::mainmenu_single_player() /home/ajenbo/code/diablo/devilutionX/Source/mainmenu.cpp:133
    #7 0x55be30101ae9 in dvl::mainmenu_loop() /home/ajenbo/code/diablo/devilutionX/Source/mainmenu.cpp:102
    #8 0x55be2fed605f in dvl::WinMain(void*, void*, char*, int) /home/ajenbo/code/diablo/devilutionX/Source/diablo.cpp:274
    #9 0x55be2fe4eeec in main /home/ajenbo/code/diablo/devilutionX/SourceX/main.cpp:28
    #10 0x7fa266c7db6a in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x26b6a)
    #11 0x55be2fd7e639 in _start (/home/ajenbo/code/diablo/devilutionX/build/devilutionx+0xd39639)
```

Probably some other fix is needed, or the values needs to not be corrected some where else.

Reply:
runs fine on 32bit

Reply:
Your on windows, it always has asan disabled, check cmakelist.txt 😉

This is how to write unstable programs, so not getting merged with this issue.

Reply:
Well it turns out that it's always set to off if your on windows, so doesn't matter if you set it to on, it's simply not available for windows yet.

Reply:
The ASAN could be fixed by calling DrawPanelBox that way
                       DrawPanelBox(PanBtnPos[i][0] - WIDTH_DIFF_2, PanBtnPos[i][1] - HEIGHT_DIFF-  336, PanBtnPos[i][2],
                                    PanBtnPos[i][3] + 1,
                                    PanBtnPos[i][0] + SCREEN_X - WIDTH_DIFF_2,
                                    PanBtnPos[i][1] + SCREEN_Y - HEIGHT_DIFF);